src/docs/operator-guides/migrating-from-outside-sharetribe/index.md

```diff
@@ -28,33 +32,102 @@ Intermediary is an edn (https://github.com/edn-format/edn) data format
+
 The id part of row 2-tuple is specified as a tuple of 1 to 3 elements.
 The first element is always an id attribute and identifies the row type.
 The second element can be an alias or an import id. A 3 element version
 has id attr, import id and an alias.
 
+```
+;; 1 element tuple
+[:im.stripeAccount/id]
+
+;; 2 element tuple with import id
+[:im.image/id #uuid "58afd8e1-e336-4ca4-a1e7-ff1d91856a6c"]
+
+;; 2 element tuple with alias
+[:im.user/id :user/jane]
+
+;; 3 element tuple
+[:im.listing/id #uuid "b074e697-ab0c-4746-a195-c58d73606b1f" :listing/rock-sauna]
+```
+
 Import ids can be given to rows and they should be unique for the
-marketplace and they are of type UUID. Aliases can be used to reference
-rows from other rows. Aliases are useful when the data is written by
-hand. Import ids are useful for programmatically generated / exported
-data. However, note that import ids are only used in the import phase
-and they can not be mapped to existing resource ids. Content part of row
-2-tuple is a map that is processed by tuple expansion function
+marketplace and have type [UUID](#uuid). Aliases can be used to
+reference rows from other rows. Aliases are useful when the data is
+written by hand. Import ids are useful for programmatically generated /
+exported data.
+
+Note that import ids are only used in the import phase and they can not
+be mapped to existing resource ids. TODO: EXPLAIN BETTER! Content part
+of row 2-tuple is a map that is processed by tuple expansion function
 registered for the content type.
 
-If you are not generating data by hand, it is highly recommended that
-you use the 2 element form of the id tuple. This means that you don't
-use aliases but unique uuid's throughout the file to reference entities.
+If you are not generating data by hand (TODO WHAT IS THE JUSTIFICATION
+HERE?), it is highly recommended that you use the 2 element form of the
+id tuple. This means that you don't use aliases but unique uuid's
+throughout the file to reference entities. You can read more about
+[generating UUIDs for your data here](https://www.uuidgenerator.net/api).
+
+#### Aliases
+
+If you currently use non-UUID ids in your data, you can use them to
+generate aliases for the data. Aliases must be unique across the file,
+and they cannot contain spaces.
+
+Since aliases are namespaced keywords (e.g. `:user/jane` in the
+example), the unique part cannot begin with a number. If your unique ids
+begin with a number, you can generate the aliases e.g. in the following
+pattern:
+
+```
+// type: 'user', id: 123
+alias = (type, id) =>
+// returns ':user/u123'
+return ':{type}/{type1stletter}{id}'
+```
+
+#### Referencing
+
+When you already have data that is dependent on each other, such as
+listings related to users and reviews related to listings, you can
+reference the data within the import file. You can refer to resources by
+either alias or UUID with the `#im/ref` syntax. The supported references
```

Comment:
In edn terminology, the `#im/ref` syntax is called [tagged element](https://github.com/edn-format/edn#tagged-elements).

Reply:
So here we could say "refer to resources ... with the `im/ref` tagged element".